eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -165,185 +161,230 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
-                .collect(Collectors.toMap(sve->toClusteredVariantEntity(sve).getHashedMessage(), Function.identity()));
-        List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> accessionWrappers =
-                generateNewRSIdAndGetAccessionWrappers(submittedVariantsGeneratingNewRS.values());
-        List<ClusteredVariantOperationEntity> clusteredVariantOperationEntities =
-                getClusteredVariantOperations(accessionWrappers, submittedVariantsGeneratingNewRS);
-        List<ClusteredVariantEntity> clusteredVariantEntities = submittedVariantsRetainingRS.stream()
-                .map(this::toClusteredVariantEntity)
-                .collect(Collectors.toList());
-
-        mongoTemplate.insert(clusteredVariantEntities, ClusteredVariantEntity.class);
-        mongoTemplate.insert(clusteredVariantOperationEntities,ClusteredVariantOperationEntity.class);
-
-        Map<String, Long> newlyGeneratedAccessions = accessionWrappers.stream()
-                                                                      .collect(Collectors.toMap(ac->ac.getHash(),ac->ac.getAccession()));
-        for(Map.Entry<String,SubmittedVariantEntity> entry: submittedVariantsGeneratingNewRS.entrySet()) {
-            SubmittedVariantEntity submittedVariant = entry.getValue();
-            Long newAccession = newlyGeneratedAccessions.get(entry.getKey());
-            updateSubmittedVariantsWithNewlyGeneratedRS(submittedVariant, newAccession, SubmittedVariantEntity.class,
-                                                        SubmittedVariantOperationEntity.class);
-            updateSubmittedVariantsWithNewlyGeneratedRS(submittedVariant, newAccession, DbsnpSubmittedVariantEntity.class,
-                                                        DbsnpSubmittedVariantOperationEntity.class);
+    private Set<String> getAllHashesForAssemblyAndRS(String assembly, Long accession) {
+        try {
+            return clusteredService.getAllByAccession(accession).stream()
+                    .filter(accWrapper -> accWrapper.getData().getAssemblyAccession().equals(assembly))
+                    .map(accWrapper -> accWrapper.getHash())
+                    .collect(Collectors.toSet());
+        } catch (AccessionMergedException | AccessionDoesNotExistException | AccessionDeprecatedException ex) {
+            logger.error("exception occurred while getting variants with accession ID {}", accession, ex);
         }
+        return Collections.emptySet();
+    }
 
-        return submittedVariantRSSplit.values().stream()
-                .flatMap(sveEntities->sveEntities.stream())
-                .collect(Collectors.toSet());
+    private List<SubmittedVariantOperationEntity> getSVOEWithMergeAndRSSplitCandidates() {
+        List<String> MERGE_AND_SPLIT_EVENTS = Arrays.asList(EventType.RS_MERGE_CANDIDATES.name(), EventType.RS_SPLIT_CANDIDATES.name());
+        Query querySubmitted = query(where("eventType").in(MERGE_AND_SPLIT_EVENTS));
```

Comment:
Query should also check for "inactiveObjects.seq" equals the remapped assembly.

Reply:
updated